[ALLUXIO-2112] Refactor property setting in HadoopUtils.java

MichaelPhantom opened the pull request:
[ALLUXIO-2112](https://alluxio.atlassian.net/browse/ALLUXIO-2112)
Guys, refactored according to my knowledge, please help review it, any issue notice me, thanks.

AmplabJenkins commented:
Can one of the admins verify this patch?

gpang commented:
ok to test

gpang commented:
Jenkins, add to whitelist

AmplabJenkins commented:
Merged build finished. Test PASSed.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/Alluxio-Pull-Request-Builder/10913/
Test PASSed.

aaudiber commented on core/client/src/main/java/alluxio/hadoop/HadoopUtils.java:
```diff
@@ -164,46 +156,26 @@ public static void addS3Credentials(Configuration conf) {
-      configuration.set(swiftSimulationConf, System.getProperty(swiftSimulationConf));
+  }
+
+  private static void setProperty(final Configuration configuration, final String propertyName) {
```
add javadoc that this method won't override existing properties even if they are set as System properties.

aaudiber commented on core/client/src/main/java/alluxio/hadoop/HadoopUtils.java:
```diff
@@ -164,46 +156,26 @@ public static void addS3Credentials(Configuration conf) {
+    setProperty(configuration, propertyNames);
+  }
+
+  private static void setProperty(final Configuration configuration, final String[] propertyNames) {
```
maybe call this `setConfigurationFromSystemProperties` and call the other method `setConfigurationFromSystemProperty`?

aaudiber commented on core/client/src/main/java/alluxio/hadoop/HadoopUtils.java:
```diff
@@ -164,46 +156,26 @@ public static void addS3Credentials(Configuration conf) {
-    if (System.getProperty(passwordKeyConf) != null
-        && configuration.get(passwordKeyConf) == null) {
-      configuration.set(passwordKeyConf, System.getProperty(passwordKeyConf));
+    final String[] propertyNames = {Constants.SWIFT_API_KEY, Constants.SWIFT_TENANT_KEY,
```
we typically don't mark local variables as `final`

aaudiber commented:
Thanks for the refactoring @MichaelPhantom! I've left a few comments in the code

MichaelPhantom commented:
@aaudiber thank you so much for the detail comments, last commit removed final decorator on local variables, renamed method names and added java document accordingly.

AmplabJenkins commented:
Merged build finished. Test PASSed.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/Alluxio-Pull-Request-Builder/10928/
Test PASSed.

aaudiber commented on core/client/src/main/java/alluxio/hadoop/HadoopUtils.java:
```diff
@@ -164,46 +156,44 @@ public static void addS3Credentials(Configuration conf) {
+   * @param configuration Hadoop configuration
+   * @param propertyNames the properties to be set
+   */
+  private static void setConfigurationFromSystemProperties(final Configuration configuration,
```
we can remove the final keyword for the method parameters as well, here and below

aaudiber commented:
@MichaelPhantom I left one last minor comment, otherwise looks good to me!

AmplabJenkins commented:
Merged build finished. Test PASSed.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/Alluxio-Pull-Request-Builder/10929/
Test PASSed.

MichaelPhantom commented:
@aaudiber the latest commit removed the final decorators as commented, thanks.

aaudiber commented:
LGTM, thanks for the iterations
